Artesian/Artesian.SDK/Factory/MarketData.cs

```diff
@@ -0,0 +1,218 @@
+        /// <remarks>
+        /// Register a MarketData
+        /// </remarks>
+        /// <param name="metadata">the entity of metadata</param>
+        /// <param name="ctk">CancellationToken</param>
+        /// <returns></returns>
+        public async Task Register(MarketDataEntity.Input metadata, CancellationToken ctk = default)
+        {
+            EnsureArg.IsNotNull(metadata, nameof(metadata));
+            EnsureArg.IsTrue(metadata.ProviderName == null || metadata.ProviderName == this.Identifier.Provider);
+            EnsureArg.IsTrue(metadata.ProviderName == null || metadata.ProviderName == this.Identifier.Provider);
+            EnsureArg.IsTrue(metadata.MarketDataName == null || metadata.MarketDataName == this.Identifier.Name);
+            EnsureArg.IsNotNullOrWhiteSpace(metadata.OriginalTimezone);
+
+            metadata.ProviderName = this.Identifier.Provider;
+            metadata.MarketDataName = this.Identifier.Name;
+
+            if (_entity != null)
+                throw new ActualTimeSerieException("Actual Time Serie is already registered with ID {0}", _entity.MarketDataId);
+
+            _entity = await _metadataService.RegisterMarketDataAsync(metadata, ctk);
+
+            Entity = new MarketDataEntityFactory(_entity);
+        }
+
+        /// <summary>
+        /// MarketData IsRegister
+        /// </summary>
+        /// <remarks>
+        /// Register a MarketData
+        /// </remarks>
+        /// <returns> true if  Marketdata si present, false if not found </returns>
+        public async Task<bool> IsRegistered(CancellationToken ctk = default)
+        {
+            if (_entity == null)
+                _entity = await _metadataService.ReadMarketDataRegistryAsync(this.Identifier, ctk);
+
+            if (_entity != null)
+            {
+                Entity = new MarketDataEntityFactory(_entity);
+                return true;
+            }
+                
+            return false;
+        }
+
+        /// <summary>
+        /// Loads MarketData Metadata
+        /// </summary>
+        /// <remarks>
+        /// Loads MarketData Metadata
+        /// </remarks>
+        /// <returns></returns>
+        public async Task LoadMetadata(CancellationToken ctk = default)
+        {
+            if (_entity == null)
+                _entity = await _metadataService.ReadMarketDataRegistryAsync(this.Identifier, ctk);
+
+            if (_entity != null)
+                Entity = new MarketDataEntityFactory(_entity);
+        }
+
+        /// <summary>
+        /// MarketData Update
+        /// </summary>
+        /// <remarks>
+        /// Update the MarketData 
+        /// </remarks>
+        /// <returns></returns>
+        public async Task Update(CancellationToken ctk = default)
+        {
+            if (_entity == null)
+                throw new ArtesianFactoryException("Market Data is not yet registered");
+
+            var marketDataEntityInput = new MarketDataEntity.Input(_entity)
```

Comment:
Why you have to copy over the fields below? The copy-ctor should have them covered already.